crates/db_views/src/custom_emoji_view.rs

```diff
@@ -0,0 +1,86 @@
+use crate::structs::CustomEmojiView;
+use diesel::{
+  result::Error,
+  BelongingToDsl,
+  ExpressionMethods,
+  JoinOnDsl,
+  NullableExpressionMethods,
+  QueryDsl,
+};
+use diesel_async::RunQueryDsl;
+use lemmy_db_schema::{
+  newtypes::{CustomEmojiId, LocalSiteId},
+  schema::{custom_emoji, custom_emoji_keyword},
+  source::{custom_emoji::CustomEmoji, custom_emoji_keyword::CustomEmojiKeyword},
+  traits::ViewToVec,
+  utils::{get_conn, DbPool},
+};
+use std::collections::HashMap;
+
+type CustomEmojiTuple = (CustomEmoji, Option<CustomEmojiKeyword>);
+
+impl CustomEmojiView {
+  pub async fn get(pool: &DbPool, emoji_id: CustomEmojiId) -> Result<Self, Error> {
+    let conn = &mut get_conn(pool).await?;
+    let emoji = custom_emoji::table
+      .find(emoji_id)
+      .first::<CustomEmoji>(conn)
+      .await?;
+    let keywords = CustomEmojiKeyword::belonging_to(&emoji)
+      .load::<CustomEmojiKeyword>(conn)
+      .await?;
+
+    let view = CustomEmojiView {
+      custom_emoji: emoji,
+      keywords,
+    };
+
+    Ok(view)
+  }
+
+  pub async fn get_all(pool: &DbPool, for_local_site_id: LocalSiteId) -> Result<Vec<Self>, Error> {
+    let conn = &mut get_conn(pool).await?;
+    let emojis = custom_emoji::table
+      .filter(custom_emoji::local_site_id.eq(for_local_site_id))
+      .left_join(
+        custom_emoji_keyword::table.on(custom_emoji_keyword::custom_emoji_id.eq(custom_emoji::id)),
+      )
+      .order(custom_emoji::category)
+      .then_order_by(custom_emoji::id)
```

Comment:
Add an index for this sort just in case.

Reply:
Fixed this and two queries

Reply:
See below